### PR TITLE
Docpurge

### PR DESCRIPTION
**Pull request by tthsqe12**
I left calcium alone because there seems to be some options available there in the elementary functions sin,cos, ect.

### CODE REVIEW

**Comment by wbhart** [comment]
I agree with getting rid of docstrings which say:

>   function ==(a::T, b::T)
> Return true if a equals b.

But shouldn't Nemo be moving towards having more complete documentation, rather than less? How can I even find the list of functions Nemo implements?

**Comment by tthsqe12** [comment]
> the list of function nemo implements

When nemo extends a function in Base, I don't think we need documentation. And if is extending base in a strange way that requires documentation, maybe it should have a different name :).

Now, what about isirreducible, factor, ...? Those should be documented *once* in AA, instead of writing the same useless docstring everywhere in Nemo. I don't think the purpose of docstrings is to prove that the function exists.

**Comment by thofma** [comment]
The `log/exp` docstrings for `padic/qadic` have some information on the precision. Should we keep them?

**Comment by wbhart** [comment]
Ok, I was confused. I didn't think we implemented all Base functions in Nemo.

**Comment by tthsqe12** [comment]
The doc strings for log/exp for pqadic was the *only* questionable removal in my opinion.

**Comment by tthsqe12** [comment]
Try this with the current state of things.
```julia
julia> using Nemo
help?> factor
```
It just gives a buch of useless documentation that requests the user to refer to extenal documentation. Once nice long informative doc string in AA would be so much better.

**Comment by wbhart** [comment]
Well AA didn't have factor until recently.

And anyway, this won't help people who use online documentation. They will look in the Nemo docs and not find any factoring functionality.

**Comment by thofma** [comment]
Don't they have to look at the AA documentation for the interfaces anyway?

**Comment by wbhart** [comment]
Yeah but the interfaces are for developers. I'm talking about users.

**Comment by thofma** [comment]
Is https://nemocas.github.io/AbstractAlgebra.jl/dev/ring/ for developers? It lists the functionality that rings satisfying the interface provide.

File: docs/src/polynomial.md [hunk 1]
```diff
@@ -228,46 +222,10 @@ isreal(n)
...
-factor_squarefree(::gfp_poly)
-factor_squarefree(::fmpz_mod_poly)
-factor_squarefree(::gfp_fmpz_poly)
-factor_squarefree(::fq_poly)
-factor_squarefree(::fq_nmod_poly)
-```
+Certain polynomials can be factored (`fmpz_poly', `nmod_poly`, `gfp_poly`,
+`fmpz_mod_poly`, `gfp_fmpz_poly`, `fq_poly`, `fq_nmod_poly`) and the interface
+follows the specification in AbstractAlgebra.jl. The following additional
+functions are available.
```
Comment (tthsqe12) [inline docs/src/polynomial.md:228]: > They will look in the Nemo docs and not find any factoring functionality.

I have simplied the docs into this simple paragraph.

**Comment by wbhart** [comment]
It's for users. Nemo should list these somewhere in my opinion, though not with as much detail.

**Comment by thofma** [comment]
I guess a simple link will be fine.

**Comment by tthsqe12** [comment]
So you want a formal markdown table in the docs with all of the polynomials that have the factoring interface?

**Comment by wbhart** [comment]
People don't follow links to AA from Nemo docs.

I think a table with polys that follow the factor interface could be one way of solving it. Eventually I think we also need some tables of Base and AA function provided for Nemo types as well, perhaps with links to the relevant docs for each entry in the table.

**Comment by tthsqe12** [comment]
So will add some general documentation to AA for:
- what to expect from the factoring interface
- what to expect from transcedental functions like exp operating on our inexact ring elems

**Pull request merged**